src/PrestaShopBundle/Resources/config/services/core/form/data_formatter.yml

```diff
@@ -1,7 +1,12 @@
 services:
-  _defaults:
-    public: true
 
-  PrestaShop\PrestaShop\Core\Form\IdentifiableObject\DataFormatter\BulkCombinationFormDataFormatter: ~
+  PrestaShop\PrestaShop\Core\Form\IdentifiableObject\DataFormatter\AbstractFormDataFormatter:
+    abstract: true
+    arguments:
+      - !php/const PrestaShopBundle\Form\Admin\Extension\ModifyAllShopsExtension::MODIFY_ALL_SHOPS_PREFIX
 
-  PrestaShop\PrestaShop\Core\Form\IdentifiableObject\DataFormatter\CombinationListFormDataFormatter: ~
+  PrestaShop\PrestaShop\Core\Form\IdentifiableObject\DataFormatter\BulkCombinationFormDataFormatter:
+    parent: PrestaShop\PrestaShop\Core\Form\IdentifiableObject\DataFormatter\AbstractFormDataFormatter
```

Comment:
Maybe add a default `bind`  so you don't need parent, and then you can use autowire

Reply:
https://symfony.com/doc/current/service_container.html#binding-arguments-by-name-or-type

Reply:
Hmm, interesting Besides since we have a dedicated file for formatters we can use the binding safely (no risk that another parameter with similar name uses it unexpectedly)
 So we can do:

```yml
services:
  _defaults:
    bind:
      $modifyAllNamePrefix: !php/const PrestaShopBundle\Form\Admin\Extension\ModifyAllShopsExtension::MODIFY_ALL_SHOPS_PREFIX
```